src/Drawables/DrawableText.vala

```diff
@@ -0,0 +1,132 @@
+ * You should have received a copy of the GNU General Public License
+ * along with Akira. If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Authored by: Martin "mbfraga" Fraga <mbfraga@gmail.com>
+ *
+ * Parts of this code are taken from GooCanvas
+ */
+
+/*
+ * Drawable for rects.
+ */
+public class Akira.Drawables.DrawableText : Drawable {
+
+    public DrawableText (double tl_x, double tl_y, double width, double height, string text) {
+       this.center_x = tl_x + width / 2.0;
+       this.center_y = tl_y + height / 2.0;
+       this.width = width;
+       this.height = height;
+       this.label = text;
+    }
+
+    public override void simple_create_path (Cairo.Context context) {
+        var w = width;
+        var h = height;
+        var x = center_x - w / 2.0;
+        var y = center_y - h / 2.0;
+
+
+        context.rectangle (x, y, w, h);
+    }
+
+    public override void paint (Cairo.Context context, Geometry.Rectangle target_bounds, double scale) {
+        // Simple bounds check
+        if (bounds.left > target_bounds.right || bounds.right < target_bounds.left
+            || bounds.top > target_bounds.bottom || bounds.bottom < target_bounds.top) {
+            return;
+        }
+
+        var w = width;
+        var h = height;
+        var x = center_x - w / 2.0;
+        var y = center_y - h / 2.0;
+
+
+        context.save ();
+
+        context.transform (transform);
+        context.set_source_rgba (0.0, 0.0, 0.0, 1.0);
+        draw_text (context, scale);
+        context.restore ();
+
+        context.new_path ();
+
+        is_drawn = true;
+    }
+
+    private void draw_text (Cairo.Context context, double scale) {
+        var m = 0;//label_margin (scale);
+
+        var w = width;
+        var h = height;
+        var x = center_x - w / 2.0;
+        var y = center_y - h / 2.0;
+
+        //var w = bounds.width - m * 2;
+        //var h = bounds.height - m * 2;
+        if (w <= 0 || h <= 0) {
+            return;
+        }
+
+        Cairo.Matrix global_transform = context.get_matrix ();
+        var tr = transform;
+        context.set_matrix (tr);
+
+        var layout = create_pango_layout (context, label, w, scale, null);
+        context.set_matrix (global_transform);
+        context.rectangle(x, y, w, h);
+        context.clip();
+        context.move_to (x, y);
+        Pango.cairo_show_layout (context, layout);
+    }
+
+    private static Pango.Layout create_pango_layout (
+        Cairo.Context context,
+        string text,
+        double layout_width,
+        double scale,
+        double* extent_width
+    ) {
+        var layout = Pango.cairo_create_layout (context);
+        var pango_context = layout.get_context ();
+
+        if (layout_width > 0) {
+            layout.set_width ( (int)(layout_width * Pango.SCALE));
+        }
+
+        layout.set_text (text, -1);
+
+        // Load font description
+        var desc = new Pango.FontDescription ();
+        desc.set_family ("Open Sans");
```

Comment:
This probably shouldn't be hardcoded, what happens if Open Sans isn't installed? You can probably get around this issue by getting this information from the parent widget's pango context.

Reply:
I assume that if open sans isn't install, the closest one will be used by pango. This item is meant to be a starting place for a lot of work that needs to go into text items (editing, etc). Ideally a font will be passed down as a component once the proper implementation is done.